src/Pinch.hs

```diff
@@ -208,6 +209,17 @@ decode :: Pinchable a => Protocol -> ByteString -> Either String a
 decode p = deserializeValue p >=> runParser . unpinch
 {-# INLINE decode #-}
 
+-- | Decode a 'Pinchable' value from the using the given 'Protocol'
+-- and also return the "unparsed" portion of the bytestring.
+--
+-- >>> let s = pack [11,0,0,0,2,0,0,0,1,97,0,0,0,1,98,0,0,0]
+-- >>> decodeWithLeftovers binaryProtocol s :: Either String [ByteString]
```

Comment:
The type hint here is incorrect because it doesn't include the leftover. I think you meant `(ByteString, [ByteString])`? The example input looks like it includes too many bytes at the start as well. For decoding just a list, you'll start with `0, 0, 0, 2` (to indicate 2 the length), followed by the elements.

To avoid confusion in the example (with `(ByteString, [ByteString])`), I would switch it to a `[Int8]`.

    let s = pack [0, 0, 0, 4, 1, 2, 3, 4]

That, decoded into `[Int8]` will give you `[1, 2, 3, 4]`.

Reply:
Hi, thank you for your quick response, I believe it's fixed now.

Type code for 8-bit integers seems to be `3` so I've used the following bytestring:
`[3,0,0,0,5,1,2,3,4,5,0,0,0]`
With decodes to:
`Right ("\NUL\NUL\NUL",[1,2,3,4,5])`

One note:
I couldn't find a way to execute these doctests and `stack test` doesn't include them so I had to do it from stack repl.

Reply:
Ah, you're right. My bad; I was wrong about the `11` prefix a well. I forgot we need the type code for the items before the length.

RE: doctests: Yeah, I don't have doctests set up. I'm not actively working on the project so I never set it up.